src/components/Orderbook.tsx

```diff
@@ -1,196 +1,414 @@
-import React, { useEffect, useState } from 'react'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Table, Header, HeaderRow, HeaderCell, Body, Row, Cell } from '@table-library/react-table-library/table'
+import { useSort, HeaderCellSort, SortToggleType } from '@table-library/react-table-library/sort'
+import * as TableTypes from '@table-library/react-table-library/types/table'
+import { useTheme } from '@table-library/react-table-library/theme'
 import * as rb from 'react-bootstrap'
-import { useTranslation } from 'react-i18next'
+import { useTranslation, TFunction } from 'react-i18next'
 import * as ObwatchApi from '../libs/JmObwatchApi'
 // @ts-ignore
 import { useSettings } from '../context/SettingsContext'
 import Balance from './Balance'
 import styles from './Orderbook.module.css'
+import Sprite from './Sprite'
 
-interface OrderbookTableProps {
-  orders: ObwatchApi.Order[]
-  maxAmountOfRows?: number
+const SORT_KEYS = {
+  type: 'TYPE',
+  counterparty: 'COUNTERPARTY',
+  fee: 'FEE',
+  minimumSize: 'MINIMUM_SIZE',
+  maximumSize: 'MAXIMUM_SIZE',
+  minerFeeContribution: 'MINER_FEE_CONTRIBUTION',
+  bondValue: 'BOND_VALUE',
 }
 
-type OrderPropName = keyof ObwatchApi.Order
+const TABLE_THEME = {
+  Table: `
+    --data-table-library_grid-template-columns: 1fr 5rem 1fr 1fr 2fr 2fr 2fr 2fr;
+    font-size: 0.9rem;
+  `,
+  BaseCell: `
+    &:nth-of-type(3) button {
+      display: flex;
+      justify-content: center;
+    }
+    &:nth-of-type(4) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(5) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(6) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(7) button {
+      display: flex;
+      justify-content: end;
+    }
+    &:nth-of-type(8) button {
+      display: flex;
+      justify-content: end;
+    }
+  `,
+  Cell: `
+    &:nth-of-type(3) {
+      text-align: center;
+    }
+    &:nth-of-type(4) {
+      text-align: right;
+    }
+    &:nth-of-type(5) {
+      text-align: right;
+    }
+    &:nth-of-type(6) {
+      text-align: right;
+    }
+    &:nth-of-type(7) {
+      text-align: right;
+    }
+    &:nth-of-type(8) {
+      text-align: right;
+    }
+  `,
+}
 
 const withTooltip = (node: React.ReactElement, tooltip: string) => {
   return (
     <rb.OverlayTrigger overlay={(props) => <rb.Tooltip {...props}>{tooltip}</rb.Tooltip>}>{node}</rb.OverlayTrigger>
   )
 }
 
-const OrderbookTable = ({ orders }: OrderbookTableProps) => {
+// `TableNode` is known to have same properties as `ObwatchApi.Order`, hence prefer casting over object destructuring
+const toOrder = (tableNode: TableTypes.TableNode): ObwatchApi.Order => tableNode as unknown as ObwatchApi.Order
```

Comment:
```suggestion
const toOrder = (tableNode: TableTypes.TableNode) => tableNode as unknown as ObwatchApi.Order
```

Just wondering: Can the TS type system not infer the return type in this case?

Reply:
It can! : D
Sometimes I am just overly verbose.

Will apply!